modules/gdnative/nativearvr/register_types.cpp

```diff
@@ -0,0 +1,427 @@
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2017 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2017 Godot Engine contributors (cf. AUTHORS.md)    */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "register_types.h"
+
+#include "arvr_interface_gdnative.h"
+#include "core/os/os.h"
+
+typedef void *void_ptr;
+typedef real_t *real_t_ptr;
+
+void arvr_call_constructor(
+		void *p_handle,
+		godot_string *p_proc_name,
+		void *p_data,
+		int p_num_args,
+		void **p_args,
+		void *r_return) {
+	if (p_handle == NULL) {
+		ERR_PRINT("No valid library handle, can't call standard varcall procedure");
+		return;
+	}
+
+	void *library_proc;
+	Error err = OS::get_singleton()->get_dynamic_library_symbol_handle(
+			p_handle,
+			*(String *)p_proc_name,
+			library_proc,
+			true); // we roll our own message
+	if (err != OK) {
+		ERR_PRINT((String("GDNative procedure \"" + *(String *)p_proc_name) + "\" does not exists and can't be called").utf8().get_data());
+		return;
+	}
+
+	void_ptr (*proc)(godot_object *);
+	proc = (void_ptr(*)(godot_object *))library_proc;
+
+	godot_object *this_object = p_args[0];
```

Comment:
This will trigger a warning because this is C++ and implicit casts between `void *` and other pointer types are not allowed. So please change that to `godot_object *this_object = (godot_object *)p_args[0];`

This is a problem through all the calling convention handlers, so please fix :)

Reply:
Ah yes, I started that and never finished it. Weirdly Apple compilers seem to be more then happy with this particular case, but it didn't like some of the other casts :)